### PR TITLE
Issue 3115 : fix javadoc warnings

### LINKED ISSUE #3115: Fix warnings generating while running ./gradlew javadocJar
**Problem description**
./gradlew javadocJar task generates many warnings which need to be fixed.

**Problem location**
Errors are generated while running javadocJar

There are 3 classes of errors

1. cannot find symbol or package does not exist - indicates missing classpath
2. Formatting issue with &lt;code&gt; section . This results in generics and lambda syntax being misinterpreted
3. Missing or incorrect elements . Eg. missing return. 

**Suggestions for an improvement**
Three problems mentioned above have following solutions

1. Classpath issue - fix the gradle script
2. Formatting issues - Use the right combination of &lt;pre&gt; {@code  and @literal
3. For missing elements, need to add them or otherwise fix it one by one.

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
**Change log description**  
Fixes build warnings/errors during javadocJar task by fixing classpath in gradle task and updating javadocs that were malformed.

**Purpose of the change**  
Fixes #3115

**What the code does**  
This is a javadoc only change, 
1. Classpath was missing in generateJavadoc task, adding it fixes most of the issues.
2. A few classes had old method signatures in `{@link }` directive, fixing these.
3. Fixes formatting issues related to code snippets containing generics and lambdas

**How to verify it**  
`./gradelw javadocJar`  should produce significantly reduced number of warnings
This is a javadoc only change there should be no functional changes. 
Also note that this change does not fix ALL warnings or errors. But it does fix majority of them. There are still a few (<25) warnings that still won't be fixed by this code as they might require some more thorough solution to work properly.

### CODE REVIEW

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java [hunk 1]
```diff
@@ -27,11 +28,11 @@
  * Conditional vs Unconditional Updates
  * * The {@link #put(String, List, Duration)} and {@link #remove(String, Collection, Duration)} methods support conditional
  * updates.
- * * If at least one of the {@link TableEntry} instances in the put() call has {@link TableEntry#getKey()#hasVersion()} true,
+ * * If at least one of the {@link TableEntry} instances in the put() call has {@link TableKey#hasVersion()} true,
  * or if at least one of the {@link TableKey} instances in the remove() call has {@link TableKey#hasVersion()} true, then
  * the operation will perform a Conditional Update. If the "hasVersion()" method returns false for ALL the items in the
  * respective collections, then the operation will perform an Unconditional Update.
- * * Conditional Updates compare the provided {@link TableKey#getVersion()} or {@link TableEntry#getKey()#getVersion()}
+ * * Conditional Updates compare the provided {@link TableKey#getVersion()} or {@link TableKey#hasVersion()}
```
Comment (andreipaduroiu) [inline segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java]: This is wrong. Revert please.

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java [hunk 2]
```diff
@@ -27,11 +28,11 @@
  * Conditional vs Unconditional Updates
  * * The {@link #put(String, List, Duration)} and {@link #remove(String, Collection, Duration)} methods support conditional
  * updates.
- * * If at least one of the {@link TableEntry} instances in the put() call has {@link TableEntry#getKey()#hasVersion()} true,
```
Comment (andreipaduroiu) [inline segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java]: The original text was correct.

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java (hunk 1 above)
Comment (andreipaduroiu) [reply segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java]: Same everywhere in this file.

File: segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java [hunk 3]
```diff
@@ -152,7 +153,7 @@
      * <ul>
      * <li>{@link StreamSegmentNotExistsException} If the Table Segment does not exist.
      * <li>{@link TableKeyTooLongException} If {@link TableKey#getKey()} exceeds {@link #maximumKeyLength()}.
-     * <li>{@link ConditionalTableUpdateException} If {@link TableKey#hasVersion()} is true and {@link TableKey#getVersion()}
+     * <li>{@link ConditionalTableUpdateException} If {@link TableKey#hasVersion()} is true and {@link TableKey#getVersion()} ()}
```
Comment (andreipaduroiu) [inline segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java]: revert

**Comment by sachin-j-joshi** [comment]
Reverted the changes made to TableStore.js

**Pull request merged**